src/transformers/training_args_tf.py

```diff
@@ -0,0 +1,75 @@
+import logging
+from dataclasses import dataclass, field
+from typing import Tuple
+
+from .file_utils import cached_property, is_tf_available, tf_required
+from .training_args import TrainingArguments
+
+
+logger = logging.getLogger(__name__)
+
+if is_tf_available():
+    import tensorflow as tf
+
+
+@dataclass
+class TFTrainingArguments(TrainingArguments):
+    optimizer_name: str = field(
+        default="adam",
+        metadata={
+            "help": 'Name of a Tensorflow optimizer among "adadelta, adagrad, adam, adamax, ftrl, nadam, rmsprop, sgd, adamw"'
+        },
+    )
+    mode: str = field(
+        default="sequence-classification",
+        metadata={"help": 'Type of task, one of "sequence-classification", "token-classification" '},
+    )
+    loss_name: str = field(
+        default="SparseCategoricalCrossentropy",
+        metadata={
+            "help": "Name of a Tensorflow loss. For the list see: https://www.tensorflow.org/api_docs/python/tf/keras/losses"
+        },
+    )
+    eval_steps: int = field(default=1000, metadata={"help": "Run an evaluation every X steps."})
+    debug: bool = field(
+        default=False, metadata={"help": "Activate the trace to record computation graphs and profiling information"}
+    )
+
+    @cached_property
+    @tf_required
+    def _setup_strategy(self) -> Tuple["tf.distribute.Strategy", int]:
+        logger.info("Tensorflow: setting up strategy")
+        gpus = tf.config.list_physical_devices("GPU")
+
+        if self.no_cuda:
+            strategy = tf.distribute.OneDeviceStrategy(device="/cpu:0")
+        else:
+            try:
+                tpu = tf.distribute.cluster_resolver.TPUClusterResolver()
+            except ValueError:
+                tpu = None
```

Comment:
A welcome sight!

Reply:
Here we check if there is a TPU cluster available, if not a ValueError is raised then we put tpu to None

I'm gonna put the same comment in the code.

Reply:
Yes, it's great to see TPU support here!